crowbar_framework/app/models/crowbar_validator.rb

```diff
@@ -98,15 +99,20 @@ def validate_hook(value, rule, path, errors)
            errors << Kwalify::ValidationError.new(msg, path)
          end
       when "IpAddress"
-         regex = /\A(?:25[0-5]|(?:2[0-4]|1\d|[1-9])?\d)(?:\.(?:25[0-5]|(?:2[0-4]|1\d|[1-9])?\d)){3}\z/
-         if value[regex].nil?
-           msg = "Should be an IP Address: #{value}"
-           errors << Kwalify::ValidationError.new(msg, path)
-         end
+        cidr = (1..128).cover? Integer(value) rescue false
```

Comment:
What if an IPv4 address is passed with a prefix larger than 32? I would think that should error out somewhere, should it be here?

Reply:
The problem is at this point, we're only validating the single input sent it. So we have no context for the other data. So here we get a string, and it is suppose to be IpAddress, this could be ipv4, ipv6, full netmask, or a cidr. So we're validating only the input for this one value, and in that context alone we have no way of knowing whether it should be range 1-32 or 1-128.

The other checks, network-json-validator and in the network barclamps we do get this context. I.E the cidr along with an IP and/or subnet so we also validate the range is the correct range for the IP version.
See: https://github.com/crowbar/crowbar/pull/2363 and https://github.com/crowbar/crowbar-core/pull/1716